core/src/main/java/jenkins/security/SecurityListener.java

```diff
@@ -78,6 +79,16 @@ protected void loggedIn(@Nonnull String username){}
      */
     protected void userCreated(@Nonnull String username) {}
 
+    /**
+     * @since TODO
+     *
+     * Fired after an existing user object property has been updated.
+     *
+     * @param p the user object property
+     * @param username the username
+     */
+    protected void userPropertyUpdated(@Nonnull UserProperty p, @Nonnull String username){}
```

Comment:
I kind of wonder if this would be more appropriate in its own `UserPropertyListener` or similar. I'm thinking in terms that it may eventually be useful to have event listeners for JobProperty, NodeProperty, and others if I'm forgetting.

Also, I'm not sure if there's precedence in this class already, but I think it might make more sense to flip the ordering of the parameters here.

Reply:
I'm not really convinced one way or another. The `userPropertyUpdated()` method doesn't quite seem to fit the other things in SecurityListener. But then, some of the other items in SecurityListener don't fit well either. The user property is more about users than specifically about security. But so are some of the other things in this file.

I guess when it comes down to it I have a slight preference for separating out a `UserPropertyListener`.

Reply:
hmh.. alright. Question though is for the new `UserPropertyListener`: it being an abstract-class versus interface ?

Reply:
@Wadeck Any thoughts on the creation of a new separate `UserPropertyListener` ?

Reply:
Separation seems better for the long term, like Matt mentioned, about the other items that could/would require such listeners in the future. And also, it's not directly security related, so not really a reason to be put in `SecurityListener`, except for the specific case of `Details`. But I imagine the goal of this listener is to be triggered everytime a property change, in general.

Reply:
@Wadeck abstract versus interface ?

Reply:
I will bet on `abstract` being the most natural way as you are providing some static helper methods (for the event trigger). The interfaces should be used for contract definition.

Reply:
Interfaces allow static methods as of Java 8. See for example the methods added to the collections API. Also default methods.